pscheduler-server/pscheduler-server/api-server/pschedulerapiserver/admin.py

```diff
@@ -111,3 +113,57 @@ def mtu_safe():
     except Exception as ex:
         log.exception()
         return error(str(ex))
+
+@application.route("/status", methods=['GET'])
+def get_status():
+    response = {}
+
+    response["time"] = str(pscheduler.datetime_as_iso8601(datetime.datetime.now()))
+
+    # query process table
+    services = {}
+    items = ["scheduler", "archiver", "ticker", "runner", "database"]
+    for proc in psutil.process_iter():
+	pinfo = proc.as_dict(attrs=['name', 'create_time'])
+        if (pinfo["name"] in items):
+            # calculate elapsed running time
+            running_time = pscheduler.seconds_as_timedelta(time.time() - pinfo["create_time"])
+            services[pinfo["name"]] = { "running": True, "time": str(pscheduler.timedelta_as_iso8601(running_time)) }
+            items.remove(pinfo["name"])
+    try:
+        # query database, calculate server run time
+        cursor = dbcursor_query("SELECT extract(epoch from current_timestamp - pg_postmaster_start_time())", onerow=True)
+    	time_val = pscheduler.seconds_as_timedelta(cursor.fetchone()[0])
+	services["database"] = { "running": True, "time": str(pscheduler.timedelta_as_iso8601(time_val))  }
+	items.remove("database")
+    except Exception as ex:
+	pass
+    
+    if len(items) > 0:
+        # there are daemons that aren't running
+        for item in items:
+            services[item] = { "running": False }
+
+    response["services"] = services    
+    
+    # query database for last run information
+    runs = {}
+    cursor = dbcursor_query("SELECT times_actual FROM run WHERE state=5")
+    times = cursor.fetchall() 
+    formatted = []
+    for val in times:
+        formatted.append(val[0].upper)
+    runs["last-finished"] = str(pscheduler.datetime_as_iso8601(max(formatted)))
```

Comment:
Server barfs on this:

```
pscheduler-api ERROR    Exception: ValueError: max() arg is an empty sequence
pscheduler-api ERROR    Traceback (most recent call last):
pscheduler-api ERROR      File "/usr/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
pscheduler-api ERROR        rv = self.dispatch_request()
pscheduler-api ERROR      File "/usr/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
pscheduler-api ERROR        return self.view_functions[rule.endpoint](**req.view_args)
pscheduler-api ERROR      File "./pschedulerapiserver/admin.py", line 156, in get_status
pscheduler-api ERROR    ValueError: max() arg is an empty sequence
```

You probably want to `SELECT max(upper(times_actual)) FROM run WHERE...`.  If there are no rows in the database, you'll get a `NULL` back.

It also occurs to me that `times_actual` isn't indexed, where `times` is.  On systems that have a lot of runs stored, this will result in an expensive linear search.  We can talk about how to optimize this when we get to the "make it run fast" stage.

Reply:
Fixed in 1bff750.